packages/apps/tools/src/components/Global/CustomTree/CustomTree.tsx

```diff
@@ -0,0 +1,224 @@
+import {
+  MonoArrowDropDown,
+  MonoArrowRight,
+  MonoCached,
+} from '@kadena/react-icons/system';
+import { Badge, Button, Stack, Text } from '@kadena/react-ui';
+import {
+  ellipsis,
+  monospaceSmallestRegular,
+  token,
+} from '@kadena/react-ui/styles';
+import classnames from 'classnames';
+import React from 'react';
+import type { ICustomAccordionProps } from '../CustomAccordion/CustomAccordion';
+import CustomAccordion from '../CustomAccordion/CustomAccordion';
+import {
+  containerStyle,
+  itemBadgeStyle,
+  itemContainerStyle,
+  itemTitleStyle,
+  reloadButtonStyles,
+} from './CustomTree.css';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type TreeItem<T> = {
+  title: string;
+  key: React.Key;
+  children: TreeItem<T>[];
+  data: T;
+  isLoading?: boolean;
+  supportsReload?: boolean;
+  label?: string | number;
+};
+
+export interface ICustomTreeProps<T>
+  extends Omit<ICustomAccordionProps<T>, 'children' | 'data'> {
+  data: TreeItem<T>[];
+  onReload: (item: TreeItem<T>) => void;
+  onItemClick: (item: TreeItem<T>) => void;
+  onExpandCollapse: (item: TreeItem<T>, expanded: boolean) => void;
+}
+
+// eslint-disable-next-line react/function-component-definition
+function CustomTree<T>({
+  data,
+  onReload,
+  onItemClick,
+  onExpandCollapse,
+  ...rest
+}: ICustomTreeProps<T>) {
+  return (
+    <CustomAccordion {...rest} data={data} className={containerStyle}>
+      {(item) => {
+        return (
+          <>
+            <Stack
+              alignItems={'center'}
+              justifyContent={'space-between'}
+              onClick={() => {
+                item.toggleExpandCollapse();
+
+                // We "toggle" the expanded state ourselves since this is still the previous state
+                onExpandCollapse(item.data, !item.isExpanded);
+              }}
```

Comment:
I thought it might be nice to not only have the toggling handled by a "small" button but also be easy to click the entire item. I can revert this of course

Reply:
I think thats the correct UX, but you don't need to put it on the "small" button as well right?

Reply:
I will keep it as-is for now and will do a UI/UX walkthrough together with Isa and alter it after if needed. Alright?